code/eyetracking/eyetrackingrun.py

```diff
@@ -0,0 +1,961 @@
+            "PupilFitMethod": pupil_fit_method,
+            "GazeMappingSettings": {
+                "CRThreshold": CR_threshold,
+                "PThreshold": pupil_threshold,
+            },
+            "CalibrationCount": calibration_count,
+            "Columns": self.samples.columns.tolist(),
+            "CalibrationType": calibration_type,
+            "CalibrationPosition": calibration_position,
+            "AverageCalibrationError": average_calibration_error,
+            "MaximalCalibrationError": max_calibration_error,
+            "EDFHeader": header,
+        }
+
+        final_info = {
+            key: value for key, value in final_info.items() if value is not None
+        }
+
+        if not final_info:
+            print("No data to write to the JSON file. Skipping.")
+            return None
+
+        if self.task_name in ("rest", "bht", "qct"):
+            output_file_name = f"func/sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_dir-{self.pe}_eyetrack.json"
+        elif self.task_name == "fixation":
+            output_file_name = f"dwi/sub-{self.participant:03d}_ses-{self.session:03d}_acq-highres_dir-{self.pe}_eyetrack.json"
+        else:
+            raise ValueError("Unknown task type")
+
+        BIDS_folder_path = Path(BIDS_folder_path)
+        output_json_path = (
+            BIDS_folder_path
+            / f"sub-{self.participant:03d}"
+            / f"ses-{self.session:03d}"
+            / output_file_name
+        )
+        output_json_path.parent.mkdir(exist_ok=True, parents=True)
+        output_json_path.write_text(json.dumps(final_info, indent=2))
+
+        return output_json_path
+
+    def plot_pupil_size(
+        self,
+        eye: str = "right",
+        notebook: bool = True,
+        save: bool = False,
+        path_save: str = ".",
+        filename: Optional[str] = None,
+    ) -> Optional[str]:
+        """
+        Plots the time series of pupil size.
+
+        Parameters
+        ----------
+        eye : str, optional
+            Specifies whether to plot for the "right" or "left" eye.
+        notebook : bool, optional
+            If True, the plot is shown in the Jupyter notebook.
+        save : bool, optional
+            If True, the plot is saved to a file.
+        path_save : str, optional
+            Path to save the plot file.
+        filename : str, optional
+            Name of the saved plot file.
+
+        Returns
+        -------
+        Optional[str]
+            The path to the saved plot file.
+
+        Example
+        -------
+        DwiSession4.plot_pupil_size(eye="left")
+        """
+        
+        if filename is None:
+            filename = f"sub-{self.participant:03d}_ses-{self.session:03d}_task-{self.task_name}_pupil_ts.pdf"
+
+        if eye == "right":
+            self.samples.pa_right[self.samples.pa_right < 1] = np.nan
+            plt.plot(self.samples["time"].values, self.samples["pa_right"].values)
```

Comment:
For all the plots, it'd be nice to see the X axis as time in seconds. To do that, you'll need to set the origin (zero) at the timestamp that is the `StartTime` in the metadata JSON file:
```suggestion
            plt.plot(self.samples["time"].values - metadata["StartTime"], self.samples["pa_right"].values)
```

Is this doable for all the plots? (you'll need to pass the metadata in some way, or at least the start time.